File: prosartisan_mobile/lib/features/marketplace/domain/entities/trade.dart
import 'package:equatable/equatable.dart';

class Trade extends Equatable {
  final int id;
  final String name;
  final int sectorId;

  const Trade({required this.id, required this.name, required this.sectorId});

  @override
  List<Object?> get props => [id, name, sectorId];
}

File: backend/app/Models/Trade.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Trade extends Model
{
    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'sector_id',
        'code',
        'name',
    ];

    /**
     * Get the sector that owns this trade.
     */
    public function sector(): BelongsTo
    {
        return $this->belongsTo(Sector::class);
    }

    /**
     * Get the artisan profiles for this trade.
     */
    public function artisanProfiles(): HasMany
    {
        return $this->hasMany(ArtisanProfile::class);
    }
}
